### PR TITLE
KAA-1136 C SDK page

### PR DESCRIPTION
**Pull request by forGGe**
Described main components as well as application field

### CODE REVIEW

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/index.md [hunk 1]
```diff
@@ -5,22 +5,84 @@ permalink: /:path/
...
+During compilation, C SDK and derived applications requires:
+ - Compiler for given target, such as `arm-none-eabi` for bare metal ARM targets.
+ - Vendor SDK, e.g. TI SDK for the CC3200 processor.
+ - java, to generate keypairs.
+
+[Nix is the prefferable way]({{root_url}}/Customization-guide/Endpoint-SDKs/C-SDK/Environment-setup/)
+to deploy build environment.
+
+Manual installation of required packages varies between targets.
+[Use links previous section](#target-support-and-portability) to find how to build and use SDK
```
Comment (RostakaGmfun) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/index.md]: `Use links from previous section`

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/index.md [hunk 2]
```diff
@@ -5,22 +5,84 @@ permalink: /:path/
 sort_idx: 10
 ---
 
-- [Intruduction](#introduction)
-- [Build configuration](#build-configuration)
+* TOC
+{:toc}
 
-# Introduction
+*C SDK* is a portable, lightweight and fast client-side tool that provides the full set of Kaa
```
Comment (akokhanovskyi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/index.md]: Please split paragraphs into lines in sentences.

Comment (akokhanovskyi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/index.md]: s/client-side tool that provides the full set of Kaa application-level features/library that provides APIs for Kaa clients to conveniently use various platform features/

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/index.md (hunk 1 above)
Comment (rasendubi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/index.md]: `from the previous`

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/index.md [hunk 3]
```diff
@@ -5,22 +5,76 @@ permalink: /:path/
...
+# Build environment
+
+During compilation, C SDK and derived applications requires:
+ - Compiler for given target, such as `arm-none-eabi` for bare metal ARM targets.
+ - Vendor SDK, e.g. TI SDK for the CC3200 processor.
+ - java, to generate keypairs.
+
+[Nix is the prefferable way]({{root_url}}/Customization-guide/Endpoint-SDKs/C-SDK/Environment-setup/) to deploy build environment.
+
+Manual installation of required packages varies between targets.
```
Comment (ohamov) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/index.md]: s/required packages/the required packages

**Pull request merged**